.githubtemplate/workflows/deploy_staging_and_production.yml

```diff
@@ -0,0 +1,82 @@
+name: Deploy Staging
+
+on:
+  # Trigger the workflow on push action in develop branch
+  # So it will trigger when the PR of the feature branch was merged.
+  push:
+    branches:
+      - develop
+jobs:
+  app_deploy:
+    name: Deploy and upload to Firebase App Distribution
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - uses: actions/checkout@v2.3.2
+
+      - name: Load Gradle cached
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches/modules-*
+            ~/.gradle/caches/jars-*
+            ~/.gradle/caches/build-cache-*
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Run Detekt
+        run: ./gradlew detekt
+
+      - name: Archive Detekt report
+        uses: actions/upload-artifact@v2
+        with:
+          name: detekt-report
+          path: build/reports/detekt/
+
+      - name: Setup timezone
+        uses: zcong1993/setup-timezone@master
+        with:
+          timezone: Asia/Bangkok
```

Comment:
these are both setup steps 🤔 it can be changed without any extra effect tho.
My thought is that: system time here is the linux time, which is bigger than the Java JVM itself, so it should be set up first, before we setup deeper into the smaller component. Any specific reason are you preferring to so Java setup first? 🤔

Reply:
oh ok I got what you meant...that makes sense too I'll move it up